src/averageAges.js

```diff
@@ -52,7 +81,36 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let count = 0;
+
+  const func = (sum, mother, index, arr) => {
+    if (mother.sex === 'f' && onlyWithSon && arr.find(person =>
+      person.mother === mother.name && person.sex === 'm')) {
+      const childrenBoys = arr.filter(person =>
+        person.mother === mother.name && person.sex === 'm');
+
+      count += childrenBoys.length;
+
+      const diff = childrenBoys.reduce((prev, child) =>
+        prev + child.born - mother.born, 0);
+
+      return sum + diff;
```

Comment:
the formatting is killing 😱
```suggestion
    if (mother.sex === 'f' && onlyWithSon 
      && arr.find(person => (
        person.mother === mother.name && person.sex === 'm'
      )
    )) {
      const childrenBoys = arr.filter(person => (
        person.mother === mother.name && person.sex === 'm'
      ));

      count += childrenBoys.length;

      const diff = childrenBoys.reduce((prev, child) => (
        prev + child.born - mother.born, 0
      ));

      return sum + diff;
```

Reply:
Исправил.